data_steward/validation/achilles.py

```diff
@@ -1,86 +1,143 @@
-
-
-def _get_load_analysis_commands(hpo_id):
-    raw_commands = sql_wrangle.get_commands(ACHILLES_DML_SQL_PATH)
-    commands = [sql_wrangle.qualify_tables(cmd, hpo_id) for cmd in raw_commands]
-    for command in commands:
-        if END_OF_IMPORTING_LOOKUP_MARKER in command.lower():
-            break
-        yield command
+INSERT_INTO = 'insert into'
 
 
 def _get_run_analysis_commands(hpo_id):
     raw_commands = sql_wrangle.get_commands(ACHILLES_DML_SQL_PATH)
     commands = [sql_wrangle.qualify_tables(cmd, hpo_id) for cmd in raw_commands]
-    i = 0
-    for command in commands:
-        if END_OF_IMPORTING_LOOKUP_MARKER in command.lower():
-            break
-        i += 1
-    return commands[i:]
+    return commands
 
 
 def load_analyses(hpo_id):
     """
     Populate achilles lookup table
-    :param hpo_id:
-    :return:
+    :param hpo_id: hpo_id of the site to run achilles on
+    :return: None
     """
-    commands = _get_load_analysis_commands(hpo_id)
-    for command in commands:
-        bq_utils.query(command)
+    project_id = app_identity.get_application_id()
+    dataset_id = bq_utils.get_dataset_id()
+    if hpo_id is None:
+        table_prefix = ""
+    else:
+        table_prefix = hpo_id + '_'
+    table_name = table_prefix + ACHILLES_ANALYSIS
+    csv_path = os.path.join(resources.resource_path, ACHILLES_ANALYSIS + '.csv')
+    json_path = os.path.join(resources.fields_path, ACHILLES_ANALYSIS + '.json')
+    with open(json_path, 'r') as f:
+        schema = json.load(f)
+    bq_utils.load_table_from_csv(project_id, dataset_id, table_name, csv_path, schema)
+
+
+def drop_or_truncate_table(command):
+    """
+    Deletes or truncates table
+    Previously, deletion was used for both truncate and drop, and this function retains the behavior
+    :param command: query to run
+    :return: None
+    """
+    if sql_wrangle.is_truncate(command):
+        table_id = sql_wrangle.get_truncate_table_name(command)
+    else:
+        table_id = sql_wrangle.get_drop_table_name(command)
+    if bq_utils.table_exists(table_id):
+        bq_utils.delete_table(table_id)
+
+
+def extract_table_id_from_query(command):
+    """
+    Returns the table_id from an insert query from achilles_dml.sql
+    :param command: command from the file 'achilles_dml.sql', could contain newlines and comments
+    :return: table_id that the command would insert to
+    :raises RuntimeError: Raised if table_id does not exist in insert query
+    """
+    table_id = None
+    for line in command.split('\n'):
+        # ignoring comments, all commands must start with 'insert into', followed by the table_id
+        if line.strip().lower().startswith(INSERT_INTO):
+            words = line.strip().split()
+            table_id = words[2]
+    if not table_id:
+        raise RuntimeError('Table does not exist for insert query %s' % command)
+    return table_id
+
+
+def convert_insert_to_append(command):
```

Comment:
What is the advantage in converting from an insert -> select? And is the insert form still being used elsewhere by other code? The dependency here on the exact formatting of the insertion SQL is a bit brittle and hard to validate, so I would lean against using this approach unless there is a strong motivation for it

Reply:
I agree, this is very fragile and can break if we change the formatting. I don't think we are using the insert form elsewhere.
BigQuery appears to be lenient with running insert statements with regards to column name mismatches and column type mismatches. I believe we were hitting the `Maximum rate of table metadata update operations` due to this. However, BQ is more stringent with `select/write append` statements, and this allowed me to make changes in the `achilles_dml.sql` and `achilles_heel_dml.sql` to cast to the correct type and rename columns to the correct name, after looking at the failed queries.
Since this process is now complete, I could revert this by removing the functions `convert_insert_to_append` and `extract_table_id_from_query`. Let me know if you believe it's better to do so.

Reply:
Yes - in my opinion I think reverting to the insert would be better, thank you!